src/main/kotlin/org/jitsi/rtp/rtp/header_extensions/SdesHeaderExtension.kt

```diff
@@ -40,17 +40,14 @@ class SdesHeaderExtension {
 
         private fun getTextValue(buf: ByteArray, offset: Int): String {
             val dataLength = getDataLengthBytes(buf, offset)
-            val copy = BufferPool.getArray(dataLength)
-            System.arraycopy(buf, offset + SdesHeaderExtension.DATA_OFFSET, copy, 0, dataLength)
-            return String(copy, 0, dataLength, StandardCharsets.US_ASCII)
+            return String(buf, offset + SdesHeaderExtension.DATA_OFFSET, dataLength, StandardCharsets.UTF_8)
         }
 
         private fun setTextValue(buf: ByteArray, offset: Int, sdesValue: String) {
             val dataLength = getDataLengthBytes(buf, offset)
             assert(dataLength == sdesValue.length) { "buffer size doesn't match SDES value length" }
-            val array = sdesValue.toByteArray(StandardCharsets.US_ASCII)
             System.arraycopy(
-                array, 0, buf,
+                sdesValue.encodeToByteArray(), 0, buf,
```

Comment:
I'm don't follow quite. `startIndex` and `endIndex` refer to range of the `sdesValue` in this case and returns an array, so it doesn't allow me to specify where to write the String to.

The only alternative I found so far would be to use `buf.set()` in a loop iterating over the content of `sdesValue`, which potentially could be pretty short since we can assume that `sdesValue` usually should contain pretty short length values.

Reply:
Ah, I was confused about `encodeToByteArray`'s parameters, ok.